deku-derive/src/macros/deku_read.rs

```diff
@@ -107,9 +107,19 @@ fn emit_struct(input: &DekuData) -> Result<TokenStream, syn::Error> {
             use core::convert::TryFrom;
             use ::#crate_::DekuReader as _;
             let __deku_reader = &mut deku::reader::Reader::new(__deku_input.0);
-            if __deku_input.1 != 0 {
-                __deku_reader.skip_bits(__deku_input.1)?;
+            let bytes = __deku_input.1 / 8;
+            let bits = __deku_input.1 % 8;
+            if bytes != 0 {
+                use ::#crate_::no_std_io::Seek;
+                use ::#crate_::no_std_io::SeekFrom;
+                if let Err(e) = __deku_reader.seek(SeekFrom::Current(i64::try_from(bytes).unwrap())) {
```

Comment:
should we handle the error from `i64::try_from`? else we should `expect` instead of `unwrap`

Reply:
Yah, the `seek_*` code has the same problem.